hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -330,8 +350,12 @@ public MerkleMap<EntityNumPair, MerkleTokenRelStatus> tokenAssociations() {
 		return getChild(StateChildIndices.TOKEN_ASSOCIATIONS);
 	}
 
-	public MerkleMap<EntityNum, MerkleSchedule> scheduleTxs() {
-		return getChild(StateChildIndices.SCHEDULE_TXS);
+	public MerkleScheduledTransactions scheduleTxs() {
+		MerkleNode scheduledTxns = getChild(StateChildIndices.SCHEDULE_TXS);
+		if (scheduledTxns instanceof MerkleMap) {
```

Comment:
_(nit)_ This work-around causes [`StateInitializationFlow.runWith`](https://github.com/hashgraph/hedera-services/blob/eth-tx-interop/hedera-node/src/main/java/com/hedera/services/context/init/StateInitializationFlow.java#L88) to always report `0` saved scheduled transactions. Would it be worth allowing a `MerkleScheduledTransactions` to have a `size()` method that could report the pending size in this case?

(Since the actual number is of course logged later, during migration, this is mostly a thin justification to leave some kind of comment. 😉 )

Reply:
you mean it will always report 0 before the migration is complete?

Reply:
fixed